quartz-core/src/main/java/org/quartz/utils/PropertiesParser.java

```
@@ -222,7 +222,7 @@ public int getIntProperty(String name) throws NumberFormatException {
 
         try {
             return Integer.parseInt(val);
-        } catch (NumberFormatException nfe) {
+        } catch (NumberFormatException nfe) { //TODO
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)